main.cpp

```diff
@@ -3103,60 +3103,85 @@ void info_guts(memory_access &raw_access, void *con) {
-                        ss << "->0x" << std::hex << e.storage_address + e.size;
-                        ss << " to 0x" << std::hex << e.runtime_address;
-                        ss << "->0x" << std::hex << e.runtime_address + e.size;
-                    } else {
-                        ss << "Load 0x" << std::hex << e.storage_address;
-                        ss << "->0x" << std::hex << e.storage_address + e.size;
+                        info_pair("load map entry " + std::to_string(i), ss.str());
                     }
-                    info_pair("load map entry " + std::to_string(i), ss.str());
                 }
-            }
 
-            // Rolling Window Delta
-            auto rwd = best_block->get_item<rolling_window_delta_item>();
-            if (rwd != nullptr) {
-                info_pair("rolling window delta", hex_string(rwd->addr));
-            }
+                // Rolling Window Delta
+                auto rwd = best_block->get_item<rolling_window_delta_item>();
+                if (rwd != nullptr) {
+                    info_pair("rolling window delta", hex_string(rwd->addr));
+                }
 
-            // Vector Table
-            auto vtor = best_block->get_item<vector_table_item>();
-            if (vtor != nullptr) {
-                info_pair("vector table", hex_string(vtor->addr));
-            }
+                // Vector Table
+                auto vtor = best_block->get_item<vector_table_item>();
+                if (vtor != nullptr) {
+                    info_pair("vector table", hex_string(vtor->addr));
+                }
 
-            // Entry Point
-            auto entry_point = best_block->get_item<entry_point_item>();
-            if (entry_point != nullptr) {
-                std::stringstream ss;
-                ss << "EP " << hex_string(entry_point->ep);
-                ss << ", SP " << hex_string(entry_point->sp);
-                if (entry_point->splim_set) ss << ", SPLIM " << hex_string(entry_point->splim);
-                info_pair("entry point", ss.str());
+                // Entry Point
+                auto entry_point = best_block->get_item<entry_point_item>();
+                if (entry_point != nullptr) {
+                    std::stringstream ss;
+                    ss << "EP " << hex_string(entry_point->ep);
+                    ss << ", SP " << hex_string(entry_point->sp);
+                    if (entry_point->splim_set) ss << ", SPLIM " << hex_string(entry_point->splim);
+                    info_pair("entry point", ss.str());
+                }
             }
 
             // Hash and Sig
             if (hash_verified != none) {
-                info_pair("hash value", hash_verified == passed ? "verified" : "incorrect");
+                info_pair("hash", hash_verified == passed ? "verified" : "incorrect");
+                if (verbose_metadata) {
+                    std::shared_ptr<hash_value_item> hash_value = best_block->get_item<hash_value_item>();
```

Comment:
Does the use of `best_block` mean that it'll always display the same hash when there are multiple IMAGE_DEFs in a block-loop, each with different hashes?

Reply:
No, `best_block` here is just the current block being printed - so for multiple IMAGE_DEFs it'll print separate hashes for each in their own metadata block printout (along with their separate load maps etc)

Reply:
Ah cool - is it worth renaming the variable to `current_block` to avoid confusion?

Reply:
Thank you! :heart_eyes: